### ISSUE TITLE
GetBalance(unspentonly=true) false positive

### ISSUE THREAD

**Issue by bijakatlykkex** [MetacoSA/QBitNinja#23]
Hello,

We have some reports that GetBalance function with unspentonly flag set to true, for address 38DdqhVuqb36jjmxTbvBkiFPXKA8EmW9Ly (which has large number of outputs and unspents) , sometimes generates outputs which has been spent.

As I have studied logs, the false positives are reported for transactions(tx1) which output to the address and all of these outputs are consumed in transactions (tx2, tx3) such that all tx1, tx2 and tx3 are mined in the same block (This may be also a wrong pointer).

I will update, if was able to generate the case deterministicly..

Any help appreciated.

Can it be related to unclosed https://github.com/MetacoSA/QBitNinja/issues/19 ?

Thanks

------
1- By GetBalance I mean: https://github.com/MetacoSA/QBitNinja/blob/master/QBitNinja.Client/QBitNinjaClient.cs#L263

**Comment by NicolasDorier** [comment]
does it eventually sort out ?

If there is a double spend (user do routinely double spent by failing to properly lock), then the GetBalance call have a conflcitedOperation part. Which show other operations conflicting with another unconfirmed transaction.

**Comment by bijakatlykkex** [comment]
Thanks for the info.

Actually the outputs shown as unspent (just sometimes) are for some months ago.

**Comment by bijakatlykkex** [comment]
I have created the https://github.com/LykkeCity/Tools/tree/master/FalsePositiveTest to generate the error, but we switched from the address now, and no more large unspent outputs available. So I am not able to reproduce the possible problem now.

**Comment by NicolasDorier** [comment]
let's keep it open

**Comment by bijakatlykkex** [comment]
OK. Thanks for your attention.

**Comment by BruceBayne** [comment]
Unconfirmed transaction generate false Positive balance for me.

```
var wallet = new BitcoinPubKeyAddress("17QeH1HFQ74mtPrZzRHpyWHkHti9cVC5y9");
		var r = qb.GetBalance(wallet, true).Result;
		var xx = qb.GetBalanceSummary(wallet);
		Debug.WriteLine(xx.Result.Spendable.Amount);
```

this is incorrect behavior, money is already spent
blockchain.info show it fine

**Comment by NicolasDorier** [comment]
@BruceBayne thanks a lot, will take a look today.

**Comment by NicolasDorier** [comment]
@BruceBayne the reason for yours is because after a while (24H I think) qbitninja stop showing unconfirmed transactions, as it would result in your outputs being lock.

**Comment by BruceBayne** [comment]
Oh thanks for fast reply, but i am stuck, right now, i cant extract Spendable coins.. can provide you any information about it

**Comment by NicolasDorier** [comment]
@BruceBayne  I just created `1.0.3.45` you can then use `qbitClient.UnconfirmedExpiration=9999;` and do your request like before. This will show your "lost" transaction.

http://qbitninja-mainnet.azurewebsites.net/balances/17QeH1HFQ74mtPrZzRHpyWHkHti9cVC5y9?unconfExpiration=9999

The default is 24.

**Comment by BruceBayne** [comment]
Yes man! you made my day happy! Thank you a lot.  How i can help for your project?

**Comment by NicolasDorier** [comment]
Just by doing what you did. Identify missing feature, and either report or implement them.
Another suggestion: Creating an azure template to easily host a QBitNinja server.

**Comment by BruceBayne** [comment]
Maybe we can describe this field in documentation or something similar?

**Comment by NicolasDorier** [comment]
yeah, in the main page of qbitninja there is sample queries, should add it there.
